Zotero/Scenes/Detail/Items/Views/ItemsTableViewHandler.swift

```diff
@@ -167,26 +141,28 @@ final class ItemsTableViewHandler: NSObject {
         return UISwipeActionsConfiguration(actions: actions)
     }
 
-    // MARK: - Data source
-
     func sourceDataForCell(for key: String) -> (UIView, CGRect?) {
         let cell = self.tableView.visibleCells.first(where: { ($0 as? ItemCell)?.key == key })
         return (self.tableView, cell?.frame)
     }
 
-    private func cellAccessory(from accessory: ItemAccessory?) -> ItemCellModel.Accessory? {
-        return accessory.flatMap({ accessory -> ItemCellModel.Accessory in
-            switch accessory {
-            case .attachment(let attachment, let parentKey):
-                let (progress, error) = self.fileDownloader?.data(for: attachment.key, parentKey: parentKey, libraryId: attachment.libraryId) ?? (nil, nil)
-                return .attachment(.stateFrom(type: attachment.type, progress: progress, error: error))
-
-            case .doi:
-                return .doi
+    func reload(modifications: [IndexPath], insertions: [IndexPath], deletions: [IndexPath], updateSnapshot: () -> Void, completion: (() -> Void)? = nil) {
```

Comment:
`ItemsTableViewHandler` is used by both `ItemsViewController` and `TrashViewController`. `ItemsViewController` takes advantage of `Realm`s built in stuff, so we don't need to touch those objects, because they auto update. We just need to reload the table view appropriately. In `TrashViewController` it would make sense, but we'd just be calculating the diff again, while we get it here for free.

Reply:
I meant creating and applying the snapshot and let the system figure out the changes, which you are right, does recalculate the diff internally. Well, unless in the future there is some complexity with UI updates, or some other feature that we miss in this approach, that's fine.